pkg/scheduler/scheduler.go

```diff
@@ -276,8 +274,7 @@ func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 			return
 		}
 		log.Error(err, errCouldNotAdmitWL)
-		s.requeueAndUpdate(log, ctx, &e.Info, err.Error())
-		s.recorder.Eventf(newWorkload, corev1.EventTypeNormal, "Pending", "%s: %v", errCouldNotAdmitWL, err)
+		s.requeueAndUpdate(log, ctx, *e, fmt.Sprintf("%s: %v", errCouldNotAdmitWL, err))
```

Comment:
```suggestion
		e.inadmissibleReason = fmt.Sprintf("%s: %v", errCouldNotAdmitWL, err)
		s.requeueAndUpdate(log, ctx, *e)
```

Reply:
How about the status of entry? If it's failed because like  timeout of apiserver. This workload should be pushed back to heap directly, right? Shall we need to add the event and update status for it?

Reply:
You are right, we should update the status or produce an event in that case. And we should push to the heap directly.

I confirm that we don't produce an event if there is a bind failure in scheduler

https://github.com/kubernetes/kubernetes/blob/9984048bd5d13d73ae09250a6f93202c322f9e2a/pkg/scheduler/schedule_one.go#L793-L796

Reply:
The status in this case is currently "assumed", which I think is fine.